packages/grid/_modules_/grid/hooks/utils/useOptionsProp.ts

```diff
@@ -20,13 +22,21 @@ export function optionsReducer(
 }
 
 export function useOptionsProp(apiRef: ApiRef, props: GridComponentProps): GridOptions {
-  const options: GridOptionsProp = React.useMemo(
-    () => ({
-      ...props,
-      localeText: { ...DEFAULT_LOCALE_TEXT, ...props.localeText },
-    }),
-    [props],
-  );
+  const contextTheme: any = useTheme();
+
+  const options: GridOptionsProp = React.useMemo(() => {
+    const copiedProps = { ...props };
+    const extendedProps = getThemeProps({
+      theme: contextTheme,
+      name: 'MuiDataGrid',
+      props: copiedProps,
+    });
+
+    return {
+      ...extendedProps,
+      localeText: { ...DEFAULT_LOCALE_TEXT, ...extendedProps.localeText },
+    };
+  }, [props, contextTheme]);
```

Comment:
This solution is inconsistent with v5. I think that we should do:

```diff
diff --git a/packages/grid/data-grid/src/DataGrid.tsx b/packages/grid/data-grid/src/DataGrid.tsx
index 8d17d624..22c39050 100644
--- a/packages/grid/data-grid/src/DataGrid.tsx
+++ b/packages/grid/data-grid/src/DataGrid.tsx
@@ -36,7 +36,8 @@ export type DataGridProps = Omit<

 const MAX_PAGE_SIZE = 100;

-const DataGrid2 = React.forwardRef<HTMLDivElement, DataGridProps>(function DataGrid(props, ref) {
+const DataGrid2 = React.forwardRef<HTMLDivElement, DataGridProps>(function DataGrid(inProps, ref) {
+  const props = useThemeProps({ props: inProps, name: 'MuiDataGrid' });
   const { className, pageSize: pageSizeProp, ...other } = props;

   let pageSize = pageSizeProp;
diff --git a/packages/grid/x-grid/src/XGrid.tsx b/packages/grid/x-grid/src/XGrid.tsx
index d72a0a9e..365d269a 100644
--- a/packages/grid/x-grid/src/XGrid.tsx
+++ b/packages/grid/x-grid/src/XGrid.tsx
@@ -18,7 +18,8 @@ LicenseInfo.setReleaseInfo(RELEASE_INFO);
 export type XGridProps = Omit<GridComponentProps, 'licenseStatus'>;

 export const XGrid = React.memo(
-  React.forwardRef<HTMLDivElement, XGridProps>(function XGrid(props, ref) {
+  React.forwardRef<HTMLDivElement, XGridProps>(function XGrid(inProps, ref) {
+    const props = useThemeProps({ props: inProps, name: 'MuiDataGrid' });
     const { className, ...other } = props;
     const licenseStatus = useLicenseVerifier();

```

Reply:
I'm ok with this. In the beginning, I was thinking of having a separate hook inside the `GridComponent`, but that results in an infinite loop because of how the `useGridOptions` hook is done. I'll update the PR with the changes.

Reply:
Updated